repository.tf

```diff
@@ -94,6 +98,36 @@ resource "azuredevops_git_repository_file" "tags" {
   }
 }
 
+resource "azuredevops_git_repository_file" "nsg" {
+  count = var.vnet_config != null && var.vnet_config.nsg == true ? 1 : 0
+  repository_id = azuredevops_git_repository.landing_zone.id
+  file          = "nsg.tf"
+  content = templatefile("${path.module}/templates/nsg.tftpl", {})
+  branch              = "refs/heads/${azuredevops_git_repository_branch.init.name}"
+  commit_message      = "Add nsg.tf"
+  overwrite_on_create = true
+
+  lifecycle {
+    ignore_changes = [commit_message]
+  }
+}
+
+resource "azuredevops_git_repository_file" "nsgyaml" {
+  count = var.vnet_config != null && var.vnet_config.nsg == true ? 1 : 0
+  repository_id = azuredevops_git_repository.landing_zone.id
+  file          = "nsg.yaml"
+  content = templatefile("${path.module}/templates/nsg.yamltpl", {
+    subscription_name = data.azurerm_subscription.this.display_name
+  })
+  branch              = "refs/heads/${azuredevops_git_repository_branch.init.name}"
+  commit_message      = "Add nsg.yaml"
+  overwrite_on_create = true
+
+  lifecycle {
+    ignore_changes = [commit_message]
+  }
+}

```

Comment:
I think in the future we should use pure terraform code for creating NSG rules. To be honest I don't know why I had the idea to use a YAML file for the rules.

Terraform is more flexible, allows us to use more expressions/loops/functions to generate rules and we could outsource standard rules into a separate module if we need it later.

Unless you see a good reason to use YAML over TF I think we should use this PR to switch to TF in the future.

Reply:
Yes, TF could be more powerful than the YAML. 

In any case, until we desing a standard NSG model to use in our MALZ, and MALZApp projects, I prefer to mantenance and use, because we have some implementation and is faster for us to deploy.

Reply:
Change NSG by code as recommendation. Delete YAML file.